src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -1784,7 +1784,7 @@ def _process_decln(self, parent, symbol_table, decl, visibility_map=None):
                 symbol_table.add(sym)
 
             # The Symbol must have the interface given by the declaration
-            sym.interface = interface
+            sym.interface = interface.copy()
```

Comment:
As an alternative we could presumably allow interfaces to be shared in the PSyIR and then replace with a modified copy when we want to change one. That feels safer to me. Are there some gotchas with that approach? I'm wondering whether that should be done here in any case, just to make sure. With this implementation I'm guessing you are thinking that shared interfaces should be disallowed? - but if so how do we enforce this? Again I feel a Sergi discussion coming along at some point.

Reply:
In my mind an interface belongs to a Symbol and should not be shared. However, I think you're right about the conversation with @sergisiso as I raised similar concerns in his PR to resolve symbols in external modules and he disagreed with me :-)

Reply:
I've added another inline comment to make it clear where I think we should be copying the interface (in the visitor) to make sure we get no side effects. I think this should be done irrespective of this discussion.